### PR TITLE
drivers: eeprom: mchp: Enable low power feature

### PR DESCRIPTION
**Pull request by Manimaran-A**
Updated the driver to support low power mode

### CODE REVIEW

File: drivers/eeprom/eeprom_mchp_xec.c [hunk 1]
```diff
@@ -321,6 +365,12 @@ static const struct eeprom_driver_api eeprom_xec_api = {
 	.size = eeprom_xec_size,
 };
 
+#ifdef CONFIG_PM_DEVICE
+#define XEC_EEPROM_PM_DEV_DEFINE(i) PM_DEVICE_DT_INST_DEFINE(i, eeprom_xec_pm_action);
+#else
+#define XEC_EEPROM_PM_DEV_DEFINE(i) /* Not used */
+#endif
```
Comment (henrikbrixandersen) [inline drivers/eeprom/eeprom_mchp_xec.c:372]: `PM_DEVICE_DT_INST_DEFINE()` does not need to be wrapped in `fdef CONFIG_PM_DEVICE` as it is already a no-op if `CONFIG_PM_DEVICE` is not enabled.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: drivers/eeprom/eeprom_mchp_xec.c (hunk 1 above)
Comment (Manimaran-A) [reply drivers/eeprom/eeprom_mchp_xec.c]: Updated the code.

**Review by jvasanth1 — DISMISSED** [review]

**Comment by Manimaran-A** [comment]
@henrikbrixandersen could you help to review and approve?

File: drivers/eeprom/eeprom_mchp_xec.c [hunk 2]
```diff
@@ -14,6 +14,10 @@
 #include <zephyr/drivers/pinctrl.h>
 
 #include <zephyr/logging/log.h>
+#include <zephyr/pm/device.h>
+#ifdef CONFIG_PM_DEVICE
+#include <zephyr/pm/policy.h>
+#endif
```
Comment (henrikbrixandersen) [inline drivers/eeprom/eeprom_mchp_xec.c:20]: Please do not use conditional includes.

```suggestion
#include <zephyr/pm/device.h>
#include <zephyr/pm/policy.h>
```

File: drivers/eeprom/eeprom_mchp_xec.c [hunk 3]
```diff
@@ -321,6 +366,7 @@ static const struct eeprom_driver_api eeprom_xec_api = {
 	.size = eeprom_xec_size,
 };
 
+
```
Comment (henrikbrixandersen) [inline drivers/eeprom/eeprom_mchp_xec.c:369]: Stray newline.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Looks good, a couple of minor nits:

File: drivers/eeprom/eeprom_mchp_xec.c (hunk 2 above)
Comment (Manimaran-A) [reply drivers/eeprom/eeprom_mchp_xec.c]: Ok, CONFIG_PM_DEVICE wrapper removed.

File: drivers/eeprom/eeprom_mchp_xec.c (hunk 3 above)
Comment (Manimaran-A) [reply drivers/eeprom/eeprom_mchp_xec.c]: Removed the newline.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by jvasanth1 — APPROVED** [review]

**Pull request merged**